pkgs/by-name/dp/dpp/package.nix

```diff
@@ -0,0 +1,51 @@
+{ stdenv
+, fetchFromGitHub
+, cmake
+, libopus
+, openssl
+, zlib
+, libsodium
+, gcc
+, pkg-config
+, autoPatchelfHook
+, lib
+}:
+stdenv.mkDerivation rec {
+  pname = "dpp";
+  version = "10.0.29";
+
+  src = fetchFromGitHub {
+    owner = "brainboxdotcc";
+    repo = "DPP";
+    rev = "v${version}";
+    hash = "sha256-BJMg3MLSfb9x/2lPHITeI3SWwW1OZVUUMVltTWUcw9I=";
+  };
+
+  nativeBuildInputs =  [
+    cmake
+    libopus
+    pkg-config
+  ] ++ lib.optional (stdenv.hostPlatform.isLinux) autoPatchelfHook;
+
+  buildInputs = [
+    openssl
+    zlib
+    libsodium
+    libopus
+    pkg-config
+  ];
+
+  cmakeFlags = let
+      libExt = stdenv.hostPlatform.extensions.sharedLibrary;
+   in [
+    "-DOPENSSL_CRYPTO_LIBRARY=${openssl.out}/lib/libcrypto${libExt}"
+    "-DOPENSSL_SSL_LIBRARY=${openssl.out}/lib/libssl${libExt}"
```

Comment:
Why are these needed?

Reply:
for darwin this path expects a .dylib for x86_64 a .so
should I remove it?
I also tried it in darwin but I think more testing is required ( worked fine in my m1 mac )

Reply:
CMake should be able to figure that out automatically.